aiida/manage/configuration/options.py

```diff
@@ -44,45 +44,51 @@ def default(self) -> Any:
 
     @property
     def description(self) -> str:
-        return self._field.field_info.description
+        return self._field.description
 
     @property
     def global_only(self) -> bool:
         from .config import ProfileOptionsSchema
-        return self._name in ProfileOptionsSchema.__fields__
+        return self._name.replace('.', '__') not in ProfileOptionsSchema.model_fields
 
     def validate(self, value: Any) -> Any:
         """Validate a value
 
         :param value: The input value
-        :param cast: Attempt to cast the value to the required type
-
         :return: The output value
-        :raise: ConfigValidationError
-
+        :raise: ConfigurationError
         """
-        value, validation_error = self._field.validate(value, {}, loc=None)
+        from pydantic import ValidationError
+
+        from .config import GlobalOptionsSchema
+
+        attribute = self.name.replace('.', '__')
 
-        if validation_error:
-            raise ConfigurationError(validation_error)
+        try:
+            result = GlobalOptionsSchema.__pydantic_validator__.validate_assignment(
```

Comment:
This seems much more complex than v1 syntax to call `validate` method on the field. Can you explain it with adding a comment?

Reply:
Done. See: https://github.com/pydantic/pydantic/discussions/7367